tools/docker/scripts/ci_entrypoint.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+# author: Ole Schuett
+
+REPORT=/workspace/report.txt
+echo -n "StartDate: " > $REPORT
+date --utc --rfc-3339=seconds >> $REPORT
+
+# Get cp2k sources.
+if [ -n "${GIT_REF}" ]; then
+    echo -e "\n========== Fetching Git Commit =========="
+    cd /workspace/cp2k
+    git fetch origin "${GIT_BRANCH}"
+    git checkout "${GIT_REF}"
+    git submodule update --init --recursive
+    git --no-pager log -1 --pretty='%nCommitSHA: %H%nCommitTime: %ci%nCommitAuthor: %an%nCommitSubject: %s%n' | tee -a $REPORT
+
+elif [ -d  /mnt/cp2k ]; then
+    echo -e "\n========== Copying Changed Files =========="
+    #TODO: Exclusion rules could also come from .gitigore
+    rsync --exclude="*~"                      \
+          --exclude=".*/"                     \
+          --exclude="*.pyc"                   \
+          --exclude=/obj/                     \
+          --exclude=/lib/                     \
+          --exclude=/exe/                     \
+          --exclude=/regtesting/              \
+          --exclude=/tools/toolchain/         \
+          --executability                     \
+          --ignore-times                      \
+          --update                            \
+          --verbose                           \
+          --recursive                         \
+          --checksum                          \
+          /mnt/cp2k/  /workspace/cp2k/
+else
+    echo "Neighter GIT_REF nor /mnt/cp2k found - aborting."
+    exit 255
+fi
+
+
+# Update toolchain.
+if [ -d /opt/cp2k-toolchain ]; then
+    echo -e "\n========== Updating Toolchain =========="
+    cd /opt/cp2k-toolchain
+    rsync --exclude="*~"          \
+          --exclude=".*/"         \
+          --exclude="*.pyc"       \
+          --exclude="/build"      \
+          --exclude="/install"    \
+          --executability         \
+          --ignore-times          \
+          --update                \
+          --verbose               \
+          --recursive             \
+          --checksum              \
+          /workspace/cp2k/tools/toolchain/  /opt/cp2k-toolchain/
+
+    # shellcheck disable=SC1091
+    source /opt/cp2k-toolchain/install/setup
+    # shellcheck disable=SC2086
+    ./install_cp2k_toolchain.sh ${CP2K_TOOLCHAIN_OPTIONS}
+fi
+
+echo -e "\n========== Running Test =========="
+cd /workspace
+"$@" 2>&1 | tee -a $REPORT &  # Launch in the background.
+TEST_PID=$!
+trap '[ -z $TEST_PID] || kill $TEST_PID' SIGHUP SIGINT SIGQUIT SIGTERM
+
+# Upload preliminary report every 30s while test is running.
+while [ -e /proc/${TEST_PID} ]; do
```

Comment:
why not use the `jobs` builtin instead? Would be more portable.

Reply:
The best I could come up with is `while jobs -p | grep "${TEST_PID}" > /dev/null; do`, which I find less readable. I don't think portability matters that much as we're using Docker.

Reply:
AFAIK Docker on Mac needs special flags to get a `/proc` in a container

Reply:
plus you might be checking any PID (after PID reuse) and not necessarily the child PID, while `jobs` should give the PID only if it was spawned from this shell.

Reply:
You're absolutely right. I was actually checking the PID of the `tee` process. I just pushed a new and improved version.